scene/2d/visible_on_screen_notifier_2d.cpp

```diff
@@ -138,6 +138,9 @@ void VisibleOnScreenEnabler2D::set_enable_node_path(NodePath p_path) {
 		return;
 	}
 	enable_node_path = p_path;
+	if (enable_node_path.is_empty()) {
+		return;
+	}
 	if (is_inside_tree() && !Engine::get_singleton()->is_editor_hint()) {
 		node_id = ObjectID();
 		Node *node = get_node(enable_node_path);
```

Comment:
```suggestion
	if (is_inside_tree() && !Engine::get_singleton()->is_editor_hint()) {
		node_id = ObjectID();
		Node *node = get_node_or_null(enable_node_path);
```
To match the documentation changes, alternatively

Reply:
I'd rather not have the "node fetching" part occur at all. Which did have me notice there's another spot where this needs to be avoided. Thanks

Reply:
`get_node_or_null` exits early with `p_path.empty()` so that's not a problem here

Though note that your changes here does *not* clear `node_id` if the path is emtpy, good to keep in mind

Reply:
My thought is to completely avoid the redundant `is_inside_tree()` & `!is_editor_hint()` check as little as it may matters, but if you do believe `get_node_or_null()` is better in the long run I can give in

Reply:
It can be both, but unless the intention is to not error on *just* empty this is needed *also*

Reply:
Ah, oh my word. I just realised the misunderstanding in the PR's name. I get it now.

No, what this PR should do is to allow `NodePath()` as a valid `enable_node_path`. Prior to this PR, it would be passed just fine but then error here. The user's assumption was that, if there is an error, this is not valid, even though it kind of is. In fact, it is the one way to disable VisibleOnScreenNotifier2D/3D.

I'll try to modify the PR and commit message to avoid this confusion.

Reply:
Then the documentation should be altered to fit that it's just empty paths that do not cause an error :)